### PR TITLE
Copy PID

### PR DESCRIPTION
**Pull request by ruixiong**
1. Removed old copy to clipboard icon
2. Added popup box on hover[big screen] or click[small screen]
3. Added feedback “ID Copied” after copying and disappears after 3 sec

### CODE REVIEW

File: fs-person/fs-person.html [hunk 1]
```diff
@@ -332,15 +315,77 @@
...
+          z-index: 1;
+          bottom: 125%;
+          left: 70%;
+          margin-left: -80px;
+          box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.5);
+      }
+      .popup {
+          margin-bottom: 5px;
+      }
+      #CopyPopupDone {
```
Comment (thorntonjn) [inline fs-person/fs-person.html]: Can you get away with not using Ids ?

File: fs-person/fs-person.html [hunk 2]
```diff
@@ -332,15 +315,78 @@
...
+      /* Popup arrow */
+      .popup .popup-text::after {
+          content: "";
+          position: absolute;
+          top: 100%;
+          left: 50%;
+          margin-left: -5px;
+          border-width: 5px;
+          border-style: solid;
+          /*border-color: #f4f4f4 transparent transparent;*/
```
Comment (thorntonjn) [inline fs-person/fs-person.html]: Should we clean this up ?

File: fs-person/fs-person.html [hunk 3]
```diff
@@ -332,15 +315,70 @@
...
+          border-radius: 6px;
+          padding: 8px 0;
+          position: absolute;
+          z-index: 1;
+          bottom: 125%;
+          left: 70%;
+          margin-left: -80px;
+          box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.5);
+      }
+      .popup {
```
Comment (thorntonjn) [inline fs-person/fs-person.html]: Should we combine the popup

File: fs-person/fs-person.html [hunk 4]
```diff
@@ -557,6 +594,38 @@
       handlePortraitError: function(e) {
         this.portraitUrl = false;
       },
+      _showCopyID: function(e) {
+        var popupList = document.querySelectorAll(".copy-popup");
+        for(var i=0; i<popupList.length; i++){
```
Comment (thorntonjn) [inline fs-person/fs-person.html]: Can you switch this to an array method.

**Comment by thorntonjn** [comment]
I'll merge as soon as you bump the version.

File: fs-person/fs-person.html [hunk 5]
```diff
@@ -162,23 +162,6 @@
         width: 15px;
       }
 
-      /* Begin GitHub-like css-only accessible tooltip */
```
Comment (skye2k2) [inline fs-person/fs-person.html]: If you are removing this code, you should remove everything until: _/* End GitHub-like css-only accessible tooltip */_

File: fs-person/fs-person.html [hunk 6]
```diff
@@ -557,6 +594,38 @@
       handlePortraitError: function(e) {
         this.portraitUrl = false;
       },
+      // this comes from stackoverflow: https://stackoverflow.com/questions/985272/selecting-text-in-an-element-akin-to-highlighting-with-your-mouse
+      _showCopyID: function(e) {
+        var popupList = document.querySelectorAll(".copy-popup");
+        popupList.forEach(function(item){
```
Comment (skye2k2) [inline fs-person/fs-person.html]: This forEach appears to be completely unnecessary. What is its purpose?

File: fs-person/fs-person.html [hunk 7]
```diff
@@ -369,9 +405,10 @@
             <span data-test-person-lifespan>[[lifespan]]</span>
             <div hidden$='[[!_showPid(pid, hidePid)]]'>
               <span class="fs-person-details__separator" hidden='[[_hideSeparator(hideSeparator, pid, lifespan)]]'>&#9679;</span>
-              <span data-test-person-pid>[[pid]]</span>
-              <span class="copy-pid-container" on-click='_copyPid' class='fs-person-details__pid'>
-                <fs-icon id="CopyPIDIcon" aria-label$='[[i18n("COPY_TO_CLIPBOARD")]]' title='' class='copy-icon tooltipped' icon='copy' data-test-person-copy-pid-button></fs-icon>
+              <span class="hover-pid popup" on-click="_showCopyID" data-test-person-pid>
+                <span class="pid-to-be-copied">[[pid]]</span>
+                <span class="popup-text copy-popup" on-click='_copyPid'><span class="copy-popup-text">Copy Person ID</span></span>
```
Comment (skye2k2) [inline fs-person/fs-person.html]: We should _never_ have hard-coded English strings--update locales or use existing strings.

File: fs-person/fs-person.html [hunk 8]
```diff
@@ -565,11 +634,14 @@
...
         this.$._pidInput.select();
         document.execCommand('copy');
-        this.$.CopyPIDIcon.setAttribute("aria-label", this.i18n("COPY_TO_CLIPBOARD_COMPLETE"));
+        window.getSelection().removeAllRanges();
+        var CopyPopup = this.querySelector(".copy-popup");
+        CopyPopup.classList.toggle("show", false);
+        var CopyPopupDone = this.querySelector(".copy-popup-done");
+        CopyPopupDone.classList.toggle("show");
         setTimeout(function(){
-          that.$.CopyPIDIcon.setAttribute("aria-label", that.i18n("COPY_TO_CLIPBOARD"));
```
Comment (skye2k2) [inline fs-person/fs-person.html]: Using two popups that are always in the DOM makes it unnecessarily hard for testers to know which is currently active, as opposed to updating the text as this basically used to.

File: fs-person/fs-person.html [hunk 9]
```diff
@@ -332,15 +315,68 @@
...
+          position: relative;
+          display: inline-block;
+          margin-bottom: 5px;
+      }
+      .popup:hover .copy-popup{
+          visibility: visible;
+      }
+      /* The actual popup (appears on top) */
+      .popup .popup-text {
+          font-family: Verdana, sans-serif;
```
Comment (skye2k2) [inline fs-person/fs-person.html]: We also attempt to keep CSS properties alphabetized, to improve compression and make debugging just a little bit faster. Notable exceptions are browser-prefixed properties that need to be out of order.

File: fs-person/fs-person.html [hunk 10]
```diff
@@ -557,6 +594,38 @@
       handlePortraitError: function(e) {
         this.portraitUrl = false;
       },
+      // this comes from stackoverflow: https://stackoverflow.com/questions/985272/selecting-text-in-an-element-akin-to-highlighting-with-your-mouse
+      _showCopyID: function(e) {
```
Comment (skye2k2) [inline fs-person/fs-person.html]: No unit tests for a function with 4 if-blocks?

File: fs-person/fs-person.html (hunk 9 above)
Comment (thorntonjn) [inline fs-person/fs-person.html]: ok,  we'll alphabetize this.  We are pretty inconsistent on this in many projects but I like the idea.

File: fs-person/fs-person.html (hunk 7 above)
Comment (thorntonjn) [inline fs-person/fs-person.html]: Good catch ! We'll fix that.

File: fs-person/fs-person.html (hunk 6 above)
Comment (thorntonjn) [inline fs-person/fs-person.html]: This code ensures that all other popup boxes dissapear when you click on another one.

File: fs-person/fs-person.html (hunk 8 above)
Comment (thorntonjn) [inline fs-person/fs-person.html]: Ok, that seems possible to do.  We will revisit that.

Comment (thorntonjn) [inline fs-person/fs-person.html]: We like testers.  👍

File: fs-person/fs-person.html (hunk 6 above)
Comment (skye2k2) [inline fs-person/fs-person.html]: When they are already self-close after a second when focus is lost. This loop just makes it happen up to a second faster. Felt superfluous to me.

File: fs-person/fs-person.html (hunk 5 above)
Comment (ruixiong) [inline fs-person/fs-person.html]: I made the changes to all of your comments in this pull request https://github.com/fs-webdev/styles-wc/pull/72. Thanks for reviewing

File: fs-person/fs-person.html (hunk 6 above)
Comment (ruixiong) [inline fs-person/fs-person.html]: Thank you cliff! I had this code before I added the auto-disappear after a second. I have it removed in the new pull request.

**Pull request merged**